httpbp/client_middlewares.go

```diff
@@ -370,21 +372,54 @@ func PrometheusClientMetrics(serverSlug string) ClientMiddleware {
 // ClientBaseplateHeadersMiddleware is a middleware that forwards baseplate headers from the context to the outgoing request.
 //
 // If it detects any new baseplate headers set on the request, it will reject the request and return an error.
-func ClientBaseplateHeadersMiddleware(client string) ClientMiddleware {
+func ClientBaseplateHeadersMiddleware(client string, store SecretsStore, path string) ClientMiddleware {
+	getSigningSecret := func() *secrets.VersionedSecret {
+		secret, err := store.GetVersionedSecret(path)
+		if err != nil {
+			log.Errorf("Failed to get secret %q: %v", path, err)
```

Comment:
for new code I'd probably prefer to avoid using the `log` library and use `slog.Error` directly instead. the adapter we use to use slog as the backend of zap have to add extra callstack layers making callsite and callstacks much less accurate.